### PR TITLE
[IMP][10.0] Partner names order configurable

### PR DESCRIPTION
**Pull request by lasley**
Forward port of #227 & #333 

[IMP] Partner names order configurable
[MIG] partner_firstname: Style improvements for forward port  …
* Prefer odoo to openerp & remove data tags
* Move exceptions file out of models
* Fix string selection in xpath
* Rename res_users and add import

### CODE REVIEW

File: partner_firstname/__manifest__.py [hunk 1]
```diff
@@ -7,17 +7,19 @@
 {
     'name': 'Partner first name and last name',
     'summary': "Split first name and last name for non company partners",
-    'version': '10.0.1.0.0',
+    'version': '10.0.2.0.0',
     'author': "Camptocamp, "
               "Grupo ESOC Ingeniería de Servicios, "
```
Comment (pedrobaeza) [inline partner_firstname/__manifest__.py:12]: I'm missing here `Tecnativa`. Antonio should forget to include it in the first place.

Comment (lasley) [reply partner_firstname/__manifest__.py:12]: Added in both versions

**Review by pedrobaeza — APPROVED** [review]
Thank you!

**Review by rafaelbn — CHANGES_REQUESTED** [review]
There is an error.

Go to:
- Settings > Configuration > General settings

And:

```
Odoo server error
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 638, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 675, in dispatch
    result = self._call_function(**self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 331, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/service/model.py", line 119, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 324, in checked_call
    result = self.endpoint(*a, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 933, in __call__
    return self.method(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/http.py", line 504, in response_wrap
    response = f(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/addons/web/controllers/main.py", line 862, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/addons/web/controllers/main.py", line 854, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/api.py", line 679, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/api.py", line 664, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3272270-334-94601a/odoo/addons/base/res/res_config.py", line 508, in default_get
    res.update(getattr(self, method)(fields))
TypeError: get_default_partner_names_order() takes exactly 1 argument (2 given)
```

File: partner_firstname/models/base_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from openerp import models, fields, api
```
Comment (susport) [inline partner_firstname/models/base_config_settings.py]: Order by name the import, about the Odoo guidelines.

File: partner_firstname/models/base_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    partner_names_order = fields.Selection(
+        string="Partner names order",
+        selection="_partner_names_order_selection",
+        help="Order to compose partner fullname",
+        required=True)
+    partner_names_order_changed = fields.Boolean(
```
Comment (susport) [inline partner_firstname/models/base_config_settings.py]: In compute method not necessary att readonly=True

**Comment by lasley** [comment]
Thanks for the reviews @rafaelbn & @susport. All notes attended to, this should be good to go assuming CI passes

**Comment by rafaelbn** [comment]
Hi @lasley , something happends with the "recaltulate names" button. It doesn't appear in 2/3 options and the icon is broken.



Thanks

**Review by susport — APPROVED** [review]
👍

**Comment by lasley** [comment]
@rafaelbn - sorry I missed your comment. Icon fixed.

Regarding the missing button - the button only appears if a change needs to be made. Otherwise, it is invisible. I triggered it back and forth a few times on my local, and it seems to work 👍

**Comment by lasley** [comment]
@rafaelbn - ahhh actually I think maybe what you were seeing was a missing api.depends. I also noticed a singleton issue that may have been causing some problems. I'm pretty sure one of these two resolved the button weirdness.

**Comment by lasley** [comment]
@rafaelbn - can you please verify that I have attended to your comments?

**Comment by lasley** [comment]
Is this good to merge?

**Comment by pedrobaeza** [comment]
Please squash your commits and I'll proceed.

**Comment by lasley** [comment]
Squashed, thanks @pedrobaeza!

**Comment by b5baxter** [comment]
Is this working?  I am still getting errors with v10.

**Pull request merged**